src/data/quot.lean

```diff
@@ -218,6 +218,19 @@ noncomputable def quotient.out [s : setoid α] : quotient s → α := quot.out
 theorem quotient.mk_out [s : setoid α] (a : α) : ⟦a⟧.out ≈ a :=
 quotient.exact (quotient.out_eq _)
 
+lemma quotient.mk_eq_iff_out [s : setoid α] {x : α} {y : quotient s} :
+  ⟦x⟧ = y ↔ x ≈ quotient.out y :=
+begin
+  split,
```

Comment:
I feel there's probably a very short proof here that avoids needing to `split` and uses `quotient.eq`, if another reviewer wants to try and golf it.

Reply:
I doubt there is such a proof if your only trick is `quotient.eq`. You can start with:
```lean
  cases quotient.exists_rep y with z hz,
  rw [← hz, quotient.eq] ,
  have := quotient.mk_out z,
```
It's clear there is a way to finish the proof from here, but it won't be very short in total. Also note that this file has access to no mathlib tactic (like `rcases` for instance). I didn't dare trying to change this.